cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -510,7 +510,8 @@ public void getReportDetails(HttpServiceRequest request, HttpServiceResponder re
       Location reportDir = reportIdDir.append(LocationName.REPORT_DIR);
       // TODO: [CDAP-13290] reports should be in avro format instead of json text;
       // TODO: [CDAP-13291] need to support reading multiple report files
-      Optional<Location> reportFile = reportDir.list().stream().filter(l -> l.getName().endsWith(".json")).findFirst();
+      Optional<Location> reportFile = reportDir.list().stream()
+        .filter(l -> l.getName().startsWith("part-")).findFirst();
```

Comment:
so this is something that affects both spark1 and 2 right as this is in base module, can you add more comment on why it fails on spark1. also add better test coverage.

Reply:
comment is added